Detectors/ITSMFT/ITS/workflow/src/TrackerSpec.cxx

```diff
@@ -91,31 +67,34 @@ void TrackerDPL::run(ProcessingContext& pc)
   if (mState != 1)
     return;
 
-  auto compClusters = pc.inputs().get<const std::vector<o2::itsmft::CompClusterExt>>("compClusters");
-  auto clusters = pc.inputs().get<const std::vector<o2::itsmft::Cluster>>("clusters");
-  auto labels = pc.inputs().get<const o2::dataformats::MCTruthContainer<o2::MCCompLabel>*>("labels");
-  auto rofs = pc.inputs().get<const std::vector<o2::itsmft::ROFRecord>>("ROframes");
-  auto mc2rofs = pc.inputs().get<const std::vector<o2::itsmft::MC2ROFRecord>>("MC2ROframes");
+  auto compClusters = pc.inputs().get<const std::vector<itsmft::CompClusterExt>>("compClusters");
+  auto clusters = pc.inputs().get<const std::vector<itsmft::Cluster>>("clusters");
+  auto rofs = pc.inputs().get<const std::vector<itsmft::ROFRecord>>("ROframes");
 
   LOG(INFO) << "ITSTracker pulled " << clusters.size() << " clusters, "
-            << labels->getIndexedSize() << " MC label objects , in "
-            << rofs.size() << " RO frames and "
-            << mc2rofs.size() << " MC events";
+            << rofs.size() << " RO frames and ";
+
+  const dataformats::MCTruthContainer<MCCompLabel>* labels = mIsMC ? pc.inputs().get<const dataformats::MCTruthContainer<MCCompLabel>*>("labels").get() : nullptr;
+  const std::vector<itsmft::MC2ROFRecord> mc2rofs = mIsMC ? pc.inputs().get<const std::vector<itsmft::MC2ROFRecord>>("MC2ROframes") : std::vector<itsmft::MC2ROFRecord>();
+  if (mIsMC) {
+    LOG(INFO) << labels->getIndexedSize() << " MC label objects , in "
+              << mc2rofs.size() << " MC events";
+  }
 
-  std::vector<o2::its::TrackITS> tracks;
-  o2::dataformats::MCTruthContainer<o2::MCCompLabel> trackLabels;
-  std::vector<o2::its::TrackITS> allTracks;
-  o2::dataformats::MCTruthContainer<o2::MCCompLabel> allTrackLabels;
+  std::vector<TrackITS> tracks;
+  dataformats::MCTruthContainer<MCCompLabel> trackLabels;
+  std::vector<TrackITS> allTracks;
+  dataformats::MCTruthContainer<MCCompLabel> allTrackLabels;
 
   std::uint32_t roFrame = 0;
-  o2::its::ROframe event(0);
+  ROframe event(0);
 
   bool continuous = mGRP->isDetContinuousReadOut("ITS");
   LOG(INFO) << "ITSTracker RO: continuous=" << continuous;
 
   if (continuous) {
     for (const auto& rof : rofs) {
-      int nclUsed = o2::its::IOUtils::loadROFrameData(rof, event, &clusters, labels.get());
+      int nclUsed = IOUtils::loadROFrameData(rof, event, &clusters, labels);
```

Comment:
@mpuccio in case of ``mIsMC==false`` you pass here ``nullptr`` for labels, but the ``IOUtils::loadROFrameData`` dereferences it unconditionally:
 https://github.com/AliceO2Group/AliceO2/blob/e05861c93b9299ed176ec8f9668cbdec93ff58f3/Detectors/ITSMFT/ITS/tracking/src/IOUtils.cxx#L160

Reply:
This was fixed in this PR (see changes in IOUtils.cxx, where I put the if guard in front of these kind of statements).